store/src/java/com/zimbra/cs/util/ProxyConfGen.java

```diff
@@ -1196,10 +1196,42 @@ public String format(Object o) {
     }
 }
 
+/**
+ * ListenAddressesVar
+ * This class is intended to produce strings that are embedded inside the 'nginx.conf.web.https.default' template
+ * It is placed inside the strict server_name enforcing server block.
+ */
+class ListenAddressesVar extends ProxyConfVar {
+
+    public ListenAddressesVar(Set<String> addresses) {
+        super("listen.:addresses",
+                null, // this is a fake attribute
+                addresses,
+                ProxyConfValueType.CUSTOM,
+                ProxyConfOverride.CUSTOM,
+                "List of ip addresses nginx needs to listen to catch all unknown server names");
+    }
+
+    @Override
+    public String format(Object o) throws ProxyConfException {
+       Set<String> addresses = (Set<String>)o;
+        if (addresses.size() == 0) {
+            return "";
+        }
+        StringBuilder sb = new StringBuilder();
+        for (String addr: addresses) {
+            sb.append(String.format("${web.strict.servername}    listen%s",
```

Comment:
I'd prefer an explicit space between listen and %s.

Reply:
I did this on purpose since I wanted to right justify it by the correct number of spaces.
The second format string accomplishes this.

Reply:
This keeps the layout of the file as similar as possible.

Reply:
Unless I'm missing something, there is no explicit space.  There's the hope that we'll never have an address that is 26 characters in length, but it's easily possible to create a valid IPv6 address that is that long or even longer.
Potentially useful reference: https://stackoverflow.com/questions/166132/maximum-length-of-the-textual-representation-of-an-ipv6-address/
- I still vote for an explicit space :-)